### PR TITLE
Added new Docs system

### PR DESCRIPTION
**Pull request by nullhook**
The new folder is in `__docs__` and the build command is: `npm run build-new-docs`

### CODE REVIEW

**Comment by nullhook** [comment]
working on adding docs to `PortalGun`

File: src/__docs__/breakpoint.mdx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    {bp => bp.isLt('medium') ? <MobileView /> : <DesktopView />}
+  </BreakpointRender>
+);
+export default MyComponent;
+```
+
+Here are the outputs of all of the bp methods given these breakpoints. If you resize your browser, they will change.
+
+```
+isLt('small')false
```
Comment (brigand) [inline src/__docs__/breakpoint.mdx]: These, of course, do not change. Could we add back the live example? I assume you'd make a separate file for the documentation components (BreakpointDocsHelpers.js or something) and import/use them here.

File: src/__docs__/breakpoint.mdx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+If element a string selector it will be matched against the closest parent. If that element doesn't match, then we attempt to match it on the grandparent, and so on, until it finds a match, or reaches document.body where it gives up.
+
+If an element isn't found immediately, we schedule low priority tasks to attempt to find the element again. This will be the case on the initial render unless you pass an HTMLElement as the element property.
+
+So what do we do if we don't have an element to check the size of? We'll, that's up to you. By default, we simply don't render the child until we're able to create a valid breakpoint object. You can override this with the canRenderWithNullBp boolean prop. If set to true, and we don't have a breakpoint, we'll pass bp as null. It's up to you to do something appropriate with the lack of information.
+
+We handle all sources of the element size changing without using timers, nor requiring you to make any special effort. There's no significant performance cost in listening for element resizes.
+
+In this example, we're using a CSS animation to change the size of the element.
```
Comment (brigand) [inline src/__docs__/breakpoint.mdx]: Please add the example with the animation here.

File: src/__docs__/ctx.mdx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+```
+
+Within the children of `Ctx.Provider`, no matter how deep, we can map the context to React nodes in render by passing a render callback child to `Ctx.use` hook. The render callback will run any time the nearest parent Ctx updates.
+
+
+```javascript
+function B() {
+  const pink = Ctx.use(state => state.color);
+  return <p style={{ color: pink }}>Hello, World!</p>;
+}
```
Comment (brigand) [inline src/__docs__/ctx.mdx]: I can work on the copy after merging. We need to explain things in a bit more detail.

File: package.json [hunk 4]
```diff
@@ -7,6 +7,7 @@
     "test": "jest",
     "docs-dev": "webpack-dev-server --mode=development --port 9398 --config ./src/docs/webpack.config.js",
     "build-docs": "webpack --mode=production --config ./src/docs/webpack.config.js && node ./scripts/buildDocs.js",
+    "build-new-docs": "docz dev",
```
Comment (brigand) [inline package.json]: We should normalize all of the scripts and clean up the old scripts.

- `build-new-docs` -> `docs:start`
- add `docs:build` (producing static html)
- add `docs:publish`: `npm run docs:build && npm run gh-pages`
- update gh-pages to use the directory `docs:build` produces, but don't run it yet

**Comment by nullhook** [comment]
added styled-components for demo components

File: src/__docs__/components/BreakpointMethods.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  return <Text active={isFlashing}>{children}</Text>;
+}
+
+function BreakpointMethods({ bp }) {
+  const results = breakpoints.map(({ name }) => (
+    <div key={name}>
+      {keys.map(key => (
+        <div key={key}>
+          <code>{`${key}('${name}')`}</code>
+          <Highlight>{String(bp[key](name))}</Highlight>
```
Comment (brigand) [inline src/__docs__/components/BreakpointMethods.js]: Needs some kind of separator 
```jsx
<code>{`${key}('${name}') === `}</code>
```

<details>

<summary>Before/After</summary>


before


after


</details>

**Pull request merged**